src/test/java/rx/observables/SyncOnSubscribeTest.java

```diff
@@ -476,8 +476,14 @@ public void testConcurrentRequests() throws InterruptedException {
         final CountDownLatch l1 = new CountDownLatch(1);
         final CountDownLatch l2 = new CountDownLatch(1);
         
-        @SuppressWarnings("unchecked")
-        Action1<? super Integer> onUnSubscribe = mock(Action1.class);
+        final CountDownLatch l3 = new CountDownLatch(1);
+
+        final Action1<Object> onUnSubscribe = new Action1<Object>() {
+            @Override
+            public void call(Object t) {
+                l3.countDown();
```

Comment:
Previous code checked that `onUnsubscribe` action was called only once, this one guarantees that action will be called at least once. Is this okay or you also want to check that action was called only once?